client/app/utils/actions/submitHandlers.ts

```diff
@@ -0,0 +1,28 @@
+"use server";
+import { revalidatePath } from "next/cache";
+import { OperationsFormData } from "../components/Form/operationsForm";
+
+export const operationSubmitHandler = async (
+  formData: OperationsFormData,
+  method: "POST" | "PUT",
+) => {
+  try {
+    const response = await fetch(
+      method === "POST"
+        ? "http://localhost:8000/api/registration/operations"
+        : `http://localhost:8000/api/registration/operations/${formData.id}`,
+      {
+        method,
+        body: JSON.stringify(formData),
+      },
+    );
+    if (!response.ok) {
+      throw new Error("i broke :(");
+    }
+    revalidatePath("/operations");
+    return await response.json();
```

Comment:
Not super happy with this as it gives the front-end a promise. Thoughts on how to do it better?